odoo/tools/config.py

```diff
@@ -615,85 +638,154 @@ def _warn_deprecated_options(self):
+
+    @classmethod
+    def _check_bool(cls, option, opt, value):
+        if value.lower() in ('1', 'yes', 'true', 'on'):
+            return True
+        if value.lower() in ('0', 'no', 'false', 'off'):
+            return False
+        raise optparse.OptionValueError(
+            f"option {opt}: invalid boolean value: {value!r}"
+        )
+
+    @classmethod
+    def _check_comma(cls, option_name, option, value):
+        return list(filter(bool, map(str.strip, value.split(','))))
+
+    @classmethod
+    def _check_path(cls, option, opt, value):
+        return cls._normalize(value)
+
+    def _parse(self, option_name, value):
+        if not isinstance(value, str):
+            e = f"can only cast strings: {value!r}"
+            raise TypeError(e)
+        if value == 'None':
+            return None
+        option = self.options_index[option_name]
+        if option.action in ('store_true', 'store_false'):
+            check_func = self._check_bool
+        elif option_name == 'without_demo':  # callback
+            check_func = self._check_bool  # noqa: SIM114
+        else:
+            check_func = self.parser.option_class.TYPE_CHECKER[option.type]
+        return check_func(option, option_name, value)
+
+    @classmethod
+    def _format_string(cls, value):
+        return str(value)
+
+    @classmethod
+    def _format_list(cls, value):
+        return ','.join(filter(bool, map(str.strip, value)))
+
+    def _format(self, option_name, value):
+        option = self.options_index[option_name]
+        if option.action in ('store_true', 'store_false'):
+            format_func = self.parser.option_class.TYPE_FORMATTER['bool']
+        elif option_name == 'without_demo':  # callback
+            format_func = self._format_string
+        else:
+            format_func = self.parser.option_class.TYPE_FORMATTER[option.type]
+        return format_func(value)
+
+    def _callback_without_demo(self, option, opt, value, parser):
+        if value is None and parser.rargs and not parser.rargs[0].startswith('-'):
+            value = parser.rargs.pop(0)
+        elif value.startswith('-'):
+            parser.rargs.insert(0, value)
+        parser.values.without_demo = value not in ('', 'false', 'False', 'none', 'None')
 
     def load(self):
+        self._warn("Since 17.0, use config._load_file_options instead", DeprecationWarning, stacklevel=2)
+        self._load_default_options(self['config'])
+
+    def _load_file_options(self, rcfile):
+        self._file_options.clear()
         outdated_options_map = {
             'xmlrpc_port': 'http_port',
             'xmlrpc_interface': 'http_interface',
             'xmlrpc': 'http_enable',
         }
         p = ConfigParser.RawConfigParser()
         try:
-            p.read([self.rcfile])
+            p.read([rcfile])
             for (name,value) in p.items('options'):
                 name = outdated_options_map.get(name, name)
```

Comment:
Add a warning if the option is outdated.

Reply:
removed this bit altogether